apis/eventstream/index.yaml

```diff
@@ -66,6 +67,11 @@ get:
               value: |
                 event: attestation
                 data: {"aggregation_bits":"0x01", "signature":"0x1b66ac1fb663c9bc59509846d6ec05345bd908eda73e670af888da41af171505cc411d61252fb6cb3fa0017b679f8bb2305b26a285fa2737f175668d0dff91cc1b66ac1fb663c9bc59509846d6ec05345bd908eda73e670af888da41af171505", "data":{"slot":"1", "index":"1", "beacon_block_root":"0xcf8e0d4e9587369b2301d0790347320302cc0943d5a1884560367e8208d920f2", "source":{"epoch":"1", "root":"0xcf8e0d4e9587369b2301d0790347320302cc0943d5a1884560367e8208d920f2"}, "target":{"epoch":"1", "root":"0xcf8e0d4e9587369b2301d0790347320302cc0943d5a1884560367e8208d920f2"}}}
+            single_attestation:
```

Comment:
I wonder if it's not better to simply add the validator index as a stand-alone field here, so that both index and committee list are available to consumers - ideally, this would be introduced in a way that doesn't require downstream tooling to change.

Reply:
> ideally, this would be introduced in a way that doesn't require downstream tooling to change.

No matter if we go with https://github.com/ethereum/consensus-specs/pull/3900 or not, there will be an impact on downstream consumers since the `Attestation` is modified in electra.

For `single_attestation` it would be best to start emitting it even pre-electra (by converting `phase0.Attestation` to `SingleAttestation`) so that consumers can switch to the new topic earlier.

Overall, it seems like the cleanest solution would be to also go with https://github.com/ethereum/consensus-specs/pull/3787 and `AggregatedAttestation` (as noted in https://github.com/ethereum/consensus-specs/pull/3900#issuecomment-2383952335). We could then add two new topics `onchain_attestation` and `aggregate_attestation` and completely deprecate `attestation` post-electra.

This seems cleaner from a consumer point of view as well, you can always subscribe to multiple topics if you want to track all kind of attestations but as it is right now, the `attestation` topic always emits all 3 different types of attestations which is not great imo.

Reply:
As far as I understand it, we need access to the state in order to convert between `SingleAttestation` and `Attestation`. So emitting an `Attestation` event for this new endpoint seems to add complexity

Reply:
> access to the state in order to convert between SingleAttestation to Attestation

you only need access to shuffling but this is already required to validate `Attestation`